src/main/scala/viper/gobra/translator/encodings/TypeEncoding.scala

```diff
@@ -172,6 +173,14 @@ trait TypeEncoding extends Generator {
     */
   def addressFootprint(ctx: Context): in.Location ==> CodeWriter[vpr.Exp] = PartialFunction.empty
 
+  /**
+    * Encodes whether a value is comparable or not.
+    * If comparability is unambiguously determined by the type of the value, then Left is returned.
+    */
+  def isComparable(ctx: Context): in.Expr ==> Either[Boolean, CodeWriter[vpr.Exp]] = {
+    case exp :: t if typ(ctx).isDefinedAt(t) =>
+      Comparability.comparable(t)(ctx.lookup).toLeft(unit(withSrc(vpr.FalseLit(), exp)))
```

Comment:
Why is a `false` literal used if comparability depends on the dynamic type?

Reply:
It is just a conservativ default.